opensrp-opd/src/main/java/org/smartregister/opd/utils/JsonFormUtils.java

```diff
@@ -0,0 +1,1031 @@
+        }
+
+        OpdLibrary.getInstance().context().getEventClientRepository().getWritableDatabase()
+                .update(Utils.metadata().getTableName(), contentValues, Constants.KEY.BASE_ENTITY_ID + " = ?",
+                        new String[]{baseEntityId});
+    }
+
+    public static String locationId(AllSharedPreferences allSharedPreferences) {
+        String providerId = allSharedPreferences.fetchRegisteredANM();
+        String userLocationId = allSharedPreferences.fetchUserLocalityId(providerId);
+        if (StringUtils.isBlank(userLocationId)) {
+            userLocationId = allSharedPreferences.fetchDefaultLocalityId(providerId);
+        }
+        return userLocationId;
+    }
+
+    protected static Triple<Boolean, JSONObject, JSONArray> validateParameters(String jsonString) {
+
+        JSONObject jsonForm = toJSONObject(jsonString);
+        JSONArray fields = fields(jsonForm);
+
+        return Triple.of(jsonForm != null && fields != null, jsonForm, fields);
+    }
+
+    protected static void processGender(JSONArray fields) {
+        try {
+            //TO DO Will need re-architecting later to support more languages, perhaps update the selector widget
+
+            JSONObject genderObject = getFieldJSONObject(fields, Constants.SEX);
+            String genderValue = "";
+
+            String rawGender = genderObject.getString(JsonFormConstants.VALUE);
+            char rawGenderChar = !TextUtils.isEmpty(rawGender) ? rawGender.charAt(0) : ' ';
+            switch (rawGenderChar) {
+                case 'm':
+                case 'M':
+                    genderValue = "Male";
+                    break;
+
+                case 'f':
+                case 'F':
+                    genderValue = "Female";
+                    break;
+
+                default:
+                    break;
+
+            }
+
+            genderObject.put(Constants.KEY.VALUE, genderValue);
+        } catch (JSONException e) {
+            Timber.e(e, "JsonFormUtils --> processGender");
+        }
+    }
+
+    protected static void processLocationFields(JSONArray fields) throws JSONException {
+        for (int i = 0; i < fields.length(); i++) {
+            if (fields.getJSONObject(i).has(JsonFormConstants.TYPE) &&
+                    fields.getJSONObject(i).getString(JsonFormConstants.TYPE).equals(JsonFormConstants.TREE))
+                try {
+                    String rawValue = fields.getJSONObject(i).getString(JsonFormConstants.VALUE);
+                    JSONArray valueArray = new JSONArray(rawValue);
+                    if (valueArray.length() > 0) {
+                        String lastLocationName = valueArray.getString(valueArray.length() - 1);
+                        String lastLocationId = LocationHelper.getInstance().getOpenMrsLocationId(lastLocationName);
+                        fields.getJSONObject(i).put(JsonFormConstants.VALUE, lastLocationId);
+                    }
+                } catch (Exception e) {
```

Comment:
no need since the whichever exception it throws will extend Exception so the class name will be given e.g java.lang.IllegalArgumentException , i think we usually specify a specific exception when we want do another thing when its thrown e.g either retry or something

Reply:
Yea, I think Exception covers everything. Right? Do we want to catch everything. Even RuntimeException is a child of Exception?

It also helps with readability. We expect X, Y, Z exceptions to happen and are OK handling them in this method. Anything else should be handled by the method calling this so they know something wrong happened